### PR TITLE
Add table aws_elasticache_replication_group closes #179 #184

### LINKED ISSUE #179: Add table aws_elasticache_replication_group
**References**
https://pkg.go.dev/github.com/aws/aws-sdk-go@v1.37.24/service/elasticache#DescribeReplicationGroupsInput

### PR DESCRIPTION
**Pull request by Paulami30**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
No env file present for the current environment:  staging 
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT undefined

SETUP: tests/aws_elasticache_replication_group []

PRETEST: tests/aws_elasticache_replication_group

TEST: tests/aws_elasticache_replication_group
Running terraform
data.aws_partition.current: Refreshing state...
data.aws_caller_identity.current: Refreshing state...
data.aws_region.alternate: Refreshing state...
data.aws_region.primary: Refreshing state...
data.null_data_source.resource: Refreshing state...
aws_vpc.my_vpc: Creating...
aws_vpc.my_vpc: Still creating... [10s elapsed]
aws_vpc.my_vpc: Still creating... [20s elapsed]
aws_vpc.my_vpc: Still creating... [30s elapsed]
aws_vpc.my_vpc: Creation complete after 39s [id=vpc-05642d8c40be4f5d5]
aws_subnet.my_subnet: Creating...
aws_subnet.my_subnet: Creation complete after 8s [id=subnet-08e3c6bcf97b08d5b]
aws_elasticache_subnet_group.my_subnet_group: Creating...
aws_elasticache_subnet_group.my_subnet_group: Creation complete after 2s [id=turbottest40722]
aws_elasticache_replication_group.named_test_resource: Creating...
aws_elasticache_replication_group.named_test_resource: Still creating... [10s elapsed]
aws_elasticache_replication_group.named_test_resource: Still creating... [20s elapsed]
aws_elasticache_replication_group.named_test_resource: Still creating... [30s elapsed]
aws_elasticache_replication_group.named_test_resource: Still creating... [40s elapsed]
aws_elasticache_replication_group.named_test_resource: Still creating... [50s elapsed]
aws_elasticache_replication_group.named_test_resource: Still creating... [1m0s elapsed]
aws_elasticache_replication_group.named_test_resource: Still creating... [1m10s elapsed]
aws_elasticache_replication_group.named_test_resource: Still creating... [1m20s elapsed]
aws_elasticache_replication_group.named_test_resource: Still creating... [1m30s elapsed]
aws_elasticache_replication_group.named_test_resource: Still creating... [1m40s elapsed]
aws_elasticache_replication_group.named_test_resource: Still creating... [1m50s elapsed]
aws_elasticache_replication_group.named_test_resource: Still creating... [2m0s elapsed]
aws_elasticache_replication_group.named_test_resource: Still creating... [2m10s elapsed]
aws_elasticache_replication_group.named_test_resource: Still creating... [2m20s elapsed]
aws_elasticache_replication_group.named_test_resource: Still creating... [2m30s elapsed]
aws_elasticache_replication_group.named_test_resource: Still creating... [2m40s elapsed]
aws_elasticache_replication_group.named_test_resource: Still creating... [2m50s elapsed]
aws_elasticache_replication_group.named_test_resource: Still creating... [3m0s elapsed]
aws_elasticache_replication_group.named_test_resource: Still creating... [3m10s elapsed]
aws_elasticache_replication_group.named_test_resource: Still creating... [3m20s elapsed]
aws_elasticache_replication_group.named_test_resource: Still creating... [3m30s elapsed]
aws_elasticache_replication_group.named_test_resource: Still creating... [3m40s elapsed]
aws_elasticache_replication_group.named_test_resource: Still creating... [3m50s elapsed]
aws_elasticache_replication_group.named_test_resource: Still creating... [4m0s elapsed]
aws_elasticache_replication_group.named_test_resource: Still creating... [4m10s elapsed]
aws_elasticache_replication_group.named_test_resource: Still creating... [4m20s elapsed]
aws_elasticache_replication_group.named_test_resource: Still creating... [4m30s elapsed]
aws_elasticache_replication_group.named_test_resource: Still creating... [4m41s elapsed]
aws_elasticache_replication_group.named_test_resource: Still creating... [4m51s elapsed]
aws_elasticache_replication_group.named_test_resource: Still creating... [5m1s elapsed]
aws_elasticache_replication_group.named_test_resource: Still creating... [5m11s elapsed]
aws_elasticache_replication_group.named_test_resource: Still creating... [5m21s elapsed]
aws_elasticache_replication_group.named_test_resource: Still creating... [5m31s elapsed]
aws_elasticache_replication_group.named_test_resource: Still creating... [5m41s elapsed]
aws_elasticache_replication_group.named_test_resource: Still creating... [5m51s elapsed]
aws_elasticache_replication_group.named_test_resource: Still creating... [6m1s elapsed]
aws_elasticache_replication_group.named_test_resource: Still creating... [6m11s elapsed]
aws_elasticache_replication_group.named_test_resource: Still creating... [6m21s elapsed]
aws_elasticache_replication_group.named_test_resource: Still creating... [6m31s elapsed]
aws_elasticache_replication_group.named_test_resource: Still creating... [6m41s elapsed]
aws_elasticache_replication_group.named_test_resource: Still creating... [6m51s elapsed]
aws_elasticache_replication_group.named_test_resource: Still creating... [7m1s elapsed]
aws_elasticache_replication_group.named_test_resource: Still creating... [7m11s elapsed]
aws_elasticache_replication_group.named_test_resource: Still creating... [7m21s elapsed]
aws_elasticache_replication_group.named_test_resource: Still creating... [7m31s elapsed]
aws_elasticache_replication_group.named_test_resource: Still creating... [7m41s elapsed]
aws_elasticache_replication_group.named_test_resource: Still creating... [7m51s elapsed]
aws_elasticache_replication_group.named_test_resource: Still creating... [8m1s elapsed]
aws_elasticache_replication_group.named_test_resource: Creation complete after 8m4s [id=turbottest40722]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 4 added, 0 changed, 0 destroyed.

Outputs:

resource_aka = arn:aws:elasticache:us-east-1:352685002396:replicationgroup:turbottest40722
resource_id = turbottest40722

Running SQL query: test-get-query.sql
[
  {
    "akas": [
      "arn:aws:elasticache:us-east-1:352685002396:replicationgroup:turbottest40722"
    ],
    "automatic_failover": "enabled",
    "cache_node_type": "cache.t2.micro",
    "description": "test description",
    "replication_group_id": "turbottest40722",
    "title": "turbottest40722"
  }
]
✔ PASSED

Running SQL query: test-hydrate-query.sql
[
  {
    "akas": [
      "arn:aws:elasticache:us-east-1:352685002396:replicationgroup:turbottest40722"
    ],
    "automatic_failover": "enabled",
    "cache_node_type": "cache.t2.micro",
    "replication_group_id": "turbottest40722",
    "title": "turbottest40722"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "akas": [
      "arn:aws:elasticache:us-east-1:352685002396:replicationgroup:turbottest40722"
    ],
    "automatic_failover": "enabled",
    "cache_node_type": "cache.t2.micro",
    "description": "test description",
    "replication_group_id": "turbottest40722",
    "title": "turbottest40722"
  }
]
✔ PASSED

Running SQL query: test-notfound-query.sql
null
✔ PASSED

POSTTEST: tests/aws_elasticache_replication_group

TEARDOWN: tests/aws_elasticache_replication_group

SUMMARY:

1/1 passed.

```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
> select
  replication_group_id,
  cache_node_type,
  at_rest_encryption_enabled
from
  aws_elasticache_replication_group
where
  not at_rest_encryption_enabled;
+----------------------+-----------------+----------------------------+
| replication_group_id | cache_node_type | at_rest_encryption_enabled |
+----------------------+-----------------+----------------------------+
| test-003             | cache.t2.micro  | false                      |
| test-001             | cache.t2.micro  | false                      |
| test-002             | <null>          | false                      |
+----------------------+-----------------+----------------------------+

> select
  replication_group_id,
  cache_node_type,
  multi_az
from
  aws_elasticache_replication_group
where
  multi_az = 'disabled';
+----------------------+-----------------+----------+
| replication_group_id | cache_node_type | multi_az |
+----------------------+-----------------+----------+
| test-002             | cache.r6g.large | disabled |
+----------------------+-----------------+----------+

> select
  replication_group_id,
  snapshot_retention_limit,
  snapshot_window,
  snapshotting_cluster_id
from
  aws_elasticache_replication_group
where
  snapshot_retention_limit < 30;
+----------------------+--------------------------+-----------------+-------------------------+
| replication_group_id | snapshot_retention_limit | snapshot_window | snapshotting_cluster_id |
+----------------------+--------------------------+-----------------+-------------------------+
| test-001             | 5                        | 09:30-10:30     | <null>                  |
| test-002             | 1                        | 06:00-07:00     | <null>                  |
+----------------------+--------------------------+-----------------+-------------------------+

> select
  cache_node_type,
  count (*)
from
  aws_elasticache_replication_group
group by
  cache_node_type;
+-----------------+-------+
| cache_node_type | count |
+-----------------+-------+
| cache.r6g.large | 1     |
| cache.t2.micro  | 1     |
+-----------------+-------+

> select
  replication_group_id,
  jsonb_array_elements_text(member_clusters) as member_clusters
from
  aws_elasticache_replication_group;
+----------------------+--------------------+
| replication_group_id | member_clusters    |
+----------------------+--------------------+
| delete-me            | delete-me-0001-001 |
| delete-me            | delete-me-0001-002 |
| delete-me            | delete-me-0001-003 |
| delete-me            | delete-me-0002-001 |
| delete-me            | delete-me-0002-002 |
| delete-me            | delete-me-0002-003 |
| delete-me            | delete-me-0003-001 |
| delete-me            | delete-me-0003-002 |
| delete-me            | delete-me-0003-003 |
+----------------------+--------------------+

```
</details>

### CODE REVIEW

File: aws/table_aws_elasticache_replication_group.go [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	// List call
+	err = svc.DescribeReplicationGroupsPages(
+		&elasticache.DescribeReplicationGroupsInput{},
+		func(page *elasticache.DescribeReplicationGroupsOutput, isLast bool) bool {
+			for _, replicationGroups := range page.ReplicationGroups {
+				d.StreamListItem(ctx, replicationGroups)
```
Comment (Subhajit97) [inline aws/table_aws_elasticache_replication_group.go:186]:
```suggestion
			for _, replicationGroup := range page.ReplicationGroups {
				d.StreamListItem(ctx, replicationGroup)
```

**Review by Subhajit97 — CHANGES_REQUESTED** [review]
@Paulami30 There is a minor change, otherwise it looks good.

**Review by Subhajit97 — APPROVED** [review]
LGTM

**Review by rajmohanty17 — APPROVED** [review]
LGTM in examples

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**